staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4137,6 +4137,12 @@ type ServiceSpec struct {
 	// If this is not specified or empty, no topology constraints will be applied.
 	// +optional
 	TopologyKeys []string `json:"topologyKeys,omitempty" protobuf:"bytes,16,opt,name=topologyKeys"`
+
+	// allocateLoadBalancerNodePorts defines if NodePorts shall be automatically
+	// allocated for services with type LoadBalancer.  Default is "true". It may be
+	// set to "false" if the cluster load-balancer does not rely on NodePorts.
+	// +optional
```

Comment:
I think here we want to mention that this field will be dropped if the alpha feature gate is not enabled. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-field-in-existing-api-version for more details on this.